### PR TITLE
PS-8190 Assertion Failure buf0buf.cc:4318:page_size.equals_to(space_p…

### PR DESCRIPTION
**Pull request by rahulmalik87**
…age_size)

https://jira.percona.com/browse/PS-8190

Problem:
Crash in debug build when discard table followed by drop table.

Analysis:
As part of discard tablespace, server sets index root page to null but
does not remove AHI(Adaptive Hash Index) entries from buffer pool.It can
causes crash if there a drop after the discard tablespace. As drop
tablespace explicity remove AHI entries but these indexes root pages are
already set to null and it asserts different places in code

Fix:
drop AHI as part of discard tablespace

### CODE REVIEW

**Review by satya-bodapati — CHANGES_REQUESTED** [review]
Please also include the testcase from other duplicate bug

File: mysql-test/suite/innodb/t/discard_drop_tablespace.test [hunk 1]
```diff
@@ -0,0 +1,415 @@
+--echo #
+--echo # Bug PS#8190 ASSERT AFTER DISCARD TABLESPACE followed by Drop Table
```
Comment (satya-bodapati) [inline mysql-test/suite/innodb/t/discard_drop_tablespace.test]:
```suggestion
--echo # PS-8190 : Assertion Failure buf0buf.cc:4318:page_size.equals_to(space_page_size)
```

Comment (satya-bodapati) [reply mysql-test/suite/innodb/t/discard_drop_tablespace.test]: or edit the title on bug page

**Review by satya-bodapati — CHANGES_REQUESTED** [review]

File: mysql-test/suite/innodb/t/discard_drop_tablespace.test (hunk 1 above)
Comment (rahulmalik87) [reply mysql-test/suite/innodb/t/discard_drop_tablespace.test]: done.

**Comment by satya-bodapati** [comment]
@rahulmalik87 please add jenkins link (innodb suites are enough)

**Comment by satya-bodapati** [comment]
> @rahulmalik87 please add jenkins link (innodb suites are enough)

@rahulmalik87 Please reply to this message

**Comment by rahulmalik87** [comment]
> > @rahulmalik87 please add jenkins link (innodb suites are enough)
> 
> @rahulmalik87 Please reply to this message

https://ps80.cd.percona.com/view/8.0/job/percona-server-8.0-param/1233/testReport/

All 4 different tests are regressions

**Review by satya-bodapati — APPROVED** [review]
Please comment on other PRs that they should be closed now

**Pull request merged**